mscore/plugin/api/elements.cpp

```diff
@@ -70,6 +70,91 @@ void Note::setTpc(int val)
             set(Pid::TPC2, val);
       }
 
+//---------------------------------------------------------
+//   Note::isChildAllowed
+///   Check if element type can be a child of note.
+///   \since MuseScore 3.3.1
+//---------------------------------------------------------
+
+bool Note::isChildAllowed(Ms::ElementType elementType)
+      {
+      switch(elementType) {
+            case ElementType::NOTEHEAD:
+            case ElementType::NOTEDOT:
+            case ElementType::FINGERING:
+            case ElementType::SYMBOL:
+            case ElementType::IMAGE:
+            case ElementType::TEXT:
+            case ElementType::BEND:
+            case ElementType::TIE:
+            case ElementType::ACCIDENTAL:
+            case ElementType::TEXTLINE:
+            case ElementType::GLISSANDO:
+                  return true;
+            default:
+                  return false;
+            }
+      }
+
+
+//---------------------------------------------------------
+//   Note::add
+///   \since MuseScore 3.3.1
+//---------------------------------------------------------
+
+void Note::add(Ms::PluginAPI::Element* wrapped)
+      {
+      Ms::Element* s = wrapped ? wrapped->element() : nullptr;
+      if (s)
+            {
+            // Ensure that the object has the expected ownership
+            if (wrapped->ownership() == Ownership::SCORE) {
+                  qWarning("Note::add: Cannot add this element. The element is already part of the score.");
+                  return;        // Don't allow operation.
+                  }
+            // Score now owns the object.
+            wrapped->setOwnership(Ownership::SCORE);
+
+            addInternal(note(), s);
+            }
+      }
+
+//---------------------------------------------------------
+//   Note::addInternal
+///   \since MuseScore 3.3.1
+//---------------------------------------------------------
+
+void Note::addInternal(Ms::Note* note, Ms::Element* s)
+      {
+      // Provide parentage for element.
+      s->setParent(note);
+      s->setTrack(note->track());
+
+      if (s && isChildAllowed(s->type())) {
+            // Create undo op and add the element.
+            toScore(note->score())->undoAddElement(s);
+            }
+      else if (s) {
+            qDebug("Note::add() not impl. %s", s->name());
+            }
+      }
+
+//---------------------------------------------------------
+//   Note::remove
+///   \since MuseScore 3.3.1
+//---------------------------------------------------------
+
+void Note::remove(Ms::PluginAPI::Element* wrapped)
+      {
+      Ms::Element* s = wrapped->element();
+      if (s && isChildAllowed(s->type())) {
```

Comment:
It would better be to check whether the element being removed is actually a child of this note, like it is done in `Chord::remove()`
https://github.com/musescore/MuseScore/blob/f067d001257f1fc583cc02d4209d05a2bb8e10fa/mscore/plugin/api/elements.cpp#L217-L218